plugins/coverbrowser.koplugin/main.lua

```diff
@@ -132,15 +133,43 @@ function CoverBrowser:addToMainMenu(menu_items)
             separator = true,
         },
         {
-            text = _("Shorten home directory to Home"),
+            text = _("Shorten home directory"),
             checked_func = function() return G_reader_settings:readSetting("home_dir_display_name") end,
             callback = function()
                 if G_reader_settings:readSetting("home_dir_display_name") then
-                    G_reader_settings:delSetting("home_dir_display_name")
-                    if FileManager.instance then FileManager.instance:reinit() end
+                  G_reader_settings:delSetting("home_dir_display_name")
+                  if FileManager.instance then FileManager.instance:reinit() end
                 else
-                    G_reader_settings:saveSetting("home_dir_display_name", _("Home"))
-                    if FileManager.instance then FileManager.instance:reinit() end
+                  local home_path_dialog
+                  home_path_dialog = InputDialog:new{
+                    title = _("Home directory name"),
+                    input = "",
```

Comment:
Seems like this should be prefilled with `home_dir_display_name`? Generically speaking again.

Reply:
Due to the nature of how that works, when the option is turned off the setting is deleted by default, so the string would be empty

Reply:
Then the nature of how that works is wrong. ;-)

Reply:
I agree. Logically, there should be 2 settings:
- `shorten_home_dir = true/false`
- `home_dir_shortened_text = "user set text like ~/"` - and if not set, use `_("Home")`

(or that other solution that make all that a non issue :)

Reply:
I’d be happy to write that tomorrow if Frenzie likes that idea

Reply:
It sounds like the correct behavior, yes.

Reply:
Consider it done, will push code tomorrow